lib/components/viewers/nearby/nearby-view.tsx

```diff
@@ -0,0 +1,183 @@
+  nearby: any
+  nearbyViewCoords?: LatLonObj
+  setHighlightedLocation: (location: Location | null) => void
+  setMainPanelContent: (content: number) => void
+  viewNearby?: (pos: LatLonObj) => void
+}
+
+const getNearbyItem = (place: any) => {
+  switch (place.__typename) {
+    case 'RentalVehicle':
+      return <Vehicle key={place.id} vehicle={place} />
+    case 'Stop':
+      return <Stop showOperatorLogo stopData={place} />
+    case 'VehicleParking':
+      return <VehicleParking place={place} />
+    case 'BikeRentalStation':
+      return <RentalStation place={place} />
+    default:
+      console.warn(
+        `Received unsupported nearby place type: ${place.__typename} `
+      )
+      return null
+  }
+}
+
+function NearbyView(props: Props): JSX.Element {
+  const {
+    fetchNearby,
+    getCurrentPosition: getPosition,
+    mobile,
+    nearby,
+    nearbyViewCoords,
+    setHighlightedLocation,
+    setMainPanelContent,
+    viewNearby
+  } = props
+  const map = useMap().current
+  const intl = useIntl()
+  const [loading, setLoading] = useState(true)
+  const firstItemRef = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    firstItemRef.current?.scrollIntoView({ behavior: 'smooth' })
+    if (nearbyViewCoords) {
+      fetchNearby(nearbyViewCoords, map)
+      setLoading(true)
+      const interval = setInterval(() => {
+        fetchNearby(nearbyViewCoords, map)
+        setLoading(true)
+      }, AUTO_REFRESH_INTERVAL)
+      return function cleanup() {
+        clearInterval(interval)
+      }
+    } else {
+      if (getPosition && viewNearby) {
+        getPosition(intl, null, (pos) => {
+          viewNearby({ lat: pos.coords.latitude, lon: pos.coords.longitude })
+        })
+      }
+    }
+  }, [nearbyViewCoords])
+
+  const onMouseEnter = useCallback(
+    (location: Location) => {
+      setHighlightedLocation(location)
+    },
+    [setHighlightedLocation]
+  )
+  const onMouseLeave = useCallback(() => {
+    setHighlightedLocation(null)
+  }, [setHighlightedLocation])
+
+  const nearbyItemList = nearby?.map((n: any) => (
+    <div
+      key={n.place.id}
+      onMouseEnter={() => onMouseEnter(n.place)}
+      onMouseLeave={onMouseLeave}
+      role="menuitem"
+      tabIndex={0}
+    >
+      {getNearbyItem(n.place)}
+    </div>
+  ))
+  useEffect(() => {
+    setLoading(false)
+  }, [nearby])
+
+  const goBack = useCallback(
+    () => setMainPanelContent(0),
+    [setMainPanelContent]
+  )
+
+  const MainContainer = mobile ? MobileContainer : Scrollable
+
+  return (
+    <MainContainer className="nearby-view base-color-bg">
+      {mobile && (
+        <MobileNavigationBar
+          headerText={intl.formatMessage({
+            id: 'components.NearbyView.header'
+          })}
+          onBackClicked={goBack}
+        />
+      )}
+      <NearbySidebarContainer
+        className="base-color-bg"
+        style={{ marginTop: mobile ? '50px' : 0 }}
+      >
+        <div ref={firstItemRef} />
```

Comment:
We need to introduce the list for screen reader users.

Reply:
What does this mean?

Reply:
It means adding a visually-hidden ~paragraph~ heading that says something like: "10 stops found near xxxxx" where xxxxx is the feature or other center of the nearby query.

Reply:
Does 7b89edf work or does it need to be outside the ol?

Reply:
Outside please.